src/middlewares/AccommodationMiddleware.js

```diff
@@ -0,0 +1,30 @@
+import AccommodationService from '../services/BookingService';
+import Response from '../helpers/Response';
+
+const isAccommodationAvail = async (req, res, next) => {
+  const { checkInDate, roomId } = req.body;
+  const serviceQueries = await AccommodationService.bookAccommodationService(req, res);
+  const {
+    isAccommodationExist, allAvailRoom, isRoomBooked
+  } = serviceQueries;
+
+  const allAvailRoomId = allAvailRoom.map(i => i.id);
+  const isRoomExist = allAvailRoomId.includes(parseInt(roomId, 10));
+  if (!isRoomExist) {
+    return Response.errorMessage(req, res, 'Room is not available', 404);
+  }
+  let isBookable;
+  isRoomBooked.forEach(x => {
+    if (new Date(x.checkOutDate) > new Date(checkInDate)) { isBookable = true; }
+  });
```

Comment:
@william000000 I think to check if a room is available for booking you should also check if someone is trying to book a room that has been booked within the check-in date and the check out date duration they are providing .

Reply:
@william000000 Great job! Just one think you could add to make your booking efficient